app/components/supplementary-file-browser.js

```diff
@@ -148,29 +153,36 @@ export default Component.extend(Analytics, {
                     action: 'click',
                     label: 'Content - Prev',
                 });
-            const start = this.get('startIndex');
-            if (start <= 0) return;
-
-            this.set('scrollAnim', `to${direction}`);
-            if ((start - 5) < 0) {
-                this.set('startIndex', 0);
-                this.set('endIndex', 6);
-            } else {
-                this.set('startIndex', start - 5);
-                this.set('endIndex', this.get('endIndex') - 5);
+
+            const startIndex = this.get('startIndex');
+
+            if (startIndex <= 0) {
+                return;
             }
+
+            const maxFilesDisplayed = this.get('maxFilesDisplayed');
+
+            this.setProperties({
+                scrollAnim: `to${direction}`,
+                startIndex: Math.max(startIndex - maxFilesDisplayed, 0),
+                endIndex: Math.max(this.get('endIndex') - maxFilesDisplayed, maxFilesDisplayed),
+            });
         },
-        changeFile(file) {
+        changeFile(selectedFile, selectedIndex) {
             this.get('metrics')
                 .trackEvent({
                     category: 'file browser',
                     action: 'select',
                     label: 'Content - File',
                 });
 
-            this.set('selectedFile', file);
-            if (this.chooseFile) {
-                this.chooseFile(file);
+            this.setProperties({
+                selectedFile,
+                selectedIndex,
+            });
+
+            if (this.attrs.chooseFile) {
+                this.sendAction('chooseFile', selectedFile); // eslint-disable-line ember/closure-actions
```

Comment:
Haven't looked at this deep enough. But is there any particular reason why we need to use `sendAction`, which is on its way to being deprecated, instead of closure actions?